File: Go/프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.go
func getGCD(a int, b int) int {
    if b == 0 {
        return a
    }
    return getGCD(b, a % b)
}

func solution(numer1 int, denom1 int, numer2 int, denom2 int) []int {
    numer := numer1 * denom2 + numer2 * denom1
    denom := denom1 * denom2
    
    gcd := getGCD(numer, denom)
    
    return []int{numer / gcd, denom / gcd}
}

File: JavaScript/프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function getGCD (a, b) {
    //a, b는 0 ~ 1000 이므로 abs를 할 필요가 없음
    return b == 0 ? a : getGCD(b, a % b);
}

function solution(numer1, denom1, numer2, denom2) {
    const numer = numer1 * denom2 + numer2 * denom1;
    const denom = denom1 * denom2;
    
    const gcd = getGCD(numer, denom);
    
    const answer = [numer / gcd, denom / gcd];
    return answer;
}